### PR TITLE
HG-113 - cleanup code to make working on these methods easier and unified where api calls are being done

### LINKED ISSUE #109: Same Events showing Cancelled and Not Cancelled in the Calendar and Event Views
@zach2825  Looks like this could be another caching or issue from importing the events reported by @pamelawoodbrowne 

[Calendar View](https://hackgreenville.com/calendar) and [Event View](https://hackgreenville.com/events)

The [API data seems to be good](https://events.openupstate.org/api/gtc).

Code For Greenville on Nov 2nd is a good example, but there appear to be others like Women Who Code.

### LINKED ISSUE #113: pull:events Import Returns an Error and Breaks Events, Could Use Some Proactive Error Checking / Validation
@zach2825 as with #109 it seemed the calendar and events lists were building up some cancellation crud. I think the root problem may be related to the events imports breaking.

The _orgs_ import still works, but I tried to clear and reimport and now we only show events for early May.

I ran a couple commands from back with our https://github.com/codeforgreenville/hackgreenville-com/issues/99 conversation to get to this current situation.
```
php artisan migrate:fresh --seed
php artisan --fix pull:events; php artisan pull:orgs;
```

The error I see on the production command line when running any version of pull:events is
image

### PR DESCRIPTION
**Pull request by zach2825**
Closes #113 
Closes #109

### CODE REVIEW

**Comment by zach2825** [comment]
I'm trying to cleanup the code for these commands to be more reliable and easier to work with. My original goal was to replace these commands and call for events on calendar, events, and org page loads. But, that would take more effort then just fixing the commands. 

I also added some tests to make sure these commands work as expected.

File: composer.json [hunk 1]
```diff
@@ -30,15 +30,14 @@
...
-        "mockery/mockery": "^1.4.4",
-        "nunomaduro/collision": "^7.0",
-        "phpunit/phpunit": "^9.0",
-        "spatie/laravel-ignition": "^2.0",
-        "roave/security-advisories": "dev-latest"
+        "laravel/telescope": "^4.14",
+        "mockery/mockery": "^1.6",
+        "nunomaduro/collision": "^7.7",
+        "pestphp/pest": "^2.8",
+        "phpunit/phpunit": "^10.2",
```
Comment (allella) [inline composer.json:39]: @zach2825 should some of these testing packages be in require-dev instead of installed on all environments?

**Review by allella — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (zach2825) [reply composer.json:39]: Yes, thank you!

**Comment by zach2825** [comment]
Even though I marked this as ready @allella can do you mind leaving this open for a few days to give other people more time to review the commands? What I did should go a long way in fixing them but the 2nd set of eyes on them would be really nice.

File: composer.json (hunk 1 above)
Comment (zach2825) [reply composer.json:39]: @allella they are in the require-dev section, lol I thought I did but wanted to double check :100:

Comment (allella) [reply composer.json:39]: My bad.

**Comment by allella** [comment]
Yeah, we can leave this open, but is there a way to patch the homepage bug in the meantime?

**Comment by zach2825** [comment]
> Yeah, we can leave this open, but is there a way to patch the homepage bug in the meantime?

Done :) @allella 
https://github.com/hackgvl/hackgreenville-com/pull/144

**Comment by allella** [comment]
@bogdankharchenko are you able to review Zach's work here with the events caching? His [question from the HackGreenville Slack](https://hackgreenville.slack.com/archives/C0420TN6W/p1688668580175679?thread_ts=1688624245.422669&cid=C0420TN6W) is posted below.

> If anybody is able to double check what I've done in this code
> https://github.com/hackgvl/hackgreenville-com/blob/deb75f01fa551d6be6c3362442a945241ba63f4c/app/Console/Commands/PullEventsCommand.php
> its supposed to load the events from the api and cache them in the database, I was going to try to factor out the command and just load the events on page load but that would be more effort then I was thinking.

**Comment by bogdankharchenko** [comment]
Let me connect with Zach before we merge this.

**Comment by allella** [comment]
Closing as #154 supersedes this PR.

**Pull request closed without merging**